### ISSUE TITLE
Editing a blog post doesn't work

### ISSUE THREAD

**Issue by godefroi** [joeaudette/cloudscribe.SimpleContent#172]
I've got the SimpleContent starter kit set up (SimpleContent+SimpleAuth+NoDb), and while I can successfully create a blog post, I cannot edit one. The editor comes up empty. If I fill the editor fields out, a new blog post is created.

**Comment by joeaudette** [comment]
It works for me.

Did that problem happen after publishing? Can you edit newly created posts?

Given that NoDb is file system storage there can be issues if you create content on one machine and then move it to another machine ie publishing it, then the file permissions on already existing files can cause a problem on the new machine, you may need to fix permissions on those files. But I would expect newly created file to be editable.

The user that is the identity on the application pool needs full control over all files and folders starting the the nodb_storage folder.

**Comment by joeaudette** [comment]
are you clicking the pencil icon on an existing post page or the plus icon which is for new posts?

**Comment by godefroi** [comment]
Yes, I'm clicking the pencil icon on an existing post. The URI of the edit page suggests that I should be editing an existing post, but the page comes up empty.

This is all on my development machine; I have not published the site or moved it anywhere.

**Comment by joeaudette** [comment]
you mentioned in another post that you changed the routing. 
The right way is to implement and inject your own [IBlogRoutes](https://github.com/joeaudette/cloudscribe.SimpleContent/blob/master/src/cloudscribe.SimpleContent.Models/IBlogRoutes.cs) so that internally the blog knows about the correct routes that you want. You can see the default [implementation here](https://github.com/joeaudette/cloudscribe.SimpleContent/blob/master/src/cloudscribe.SimpleContent.Models/DefaultBlogRoutes.cs).

make sure you add yours to services before the default one by putting it before the line for services.AddSimpleContent, like this:

    services.AddScoped<IBlogRoutes, YourCustomBlogRoutes>();
    services.AddSimpleContent();

**Comment by joeaudette** [comment]
you should also remove this line:

    routes.AddStandardRoutesForSimpleContent();

and instead add your own routes, with a similar method, you can see the code to copy and modify for your [purposes here](https://github.com/joeaudette/cloudscribe.SimpleContent/blob/master/src/cloudscribe.SimpleContent.Web.Mvc/StartupExtensions.cs)

also I hope you recently got the starter kits because there have been recent changes in the starter kits

**Comment by godefroi** [comment]
I've reverted the routes to the default, and the issue still occurs.

**Comment by joeaudette** [comment]
I cannot reproduce the problem using the latest version of that starter kit https://github.com/joeaudette/cloudscribe.StarterKits/tree/master/SimpleContent-SimpleAuth-nodb

**Comment by joeaudette** [comment]
what operating system are you on? are you using VS 2017 or VSCode?

**Comment by godefroi** [comment]
VS 2017.

**Comment by joeaudette** [comment]
are you using the latest version of that starterkit? can you reproduce the problem if you download a new copy?

**Comment by godefroi** [comment]
I can't reproduce with a newly-downloaded version. Clearly there's something I've changed here which has messed it up. Is there any sort of logging or other way to troubleshoot it?

**Comment by joeaudette** [comment]
I would first check for js errors in the page using web browser dev tools
if you run vs in debug mode you should get log output in the output window
if you use dotnet run from the command line you should get log output in the command window

**Comment by joeaudette** [comment]
I recommend copy the project to your own github or other source repository so you have good change tracking as you make changes

**Comment by joeaudette** [comment]
it is "possible" that a bug existed that I did not know about but that has been fixed in recent changes to the starter kits, so if you downloaded it a while back that could be the difference

**Comment by godefroi** [comment]
The two versions I have here (one I modified, one I didn't) were downloaded last week, at the same time (I was troubleshooting a different issue caused by my bad reading skills).

Turns out this particular issue was caused by having the "default" route mapped before any of the cloudscribe-specific routes. Once I put them in the correct order, it works!

Now, one more question (and I know this isn't the place, but...) how do I make the blog the root of the site? I'd like the blog at / and pages at /pages (or hey, at / as well). Is this sort of thing possible?
